### PR TITLE
[Merged by Bors] - Fix unsound `EntityMut::remove_children`. Add `EntityMut::world_scope`

### PR DESCRIPTION
**Pull request by devil-ira**
`EntityMut::remove_children` does not call `self.update_location()` which is unsound.
Verified by adding the following assertion, which fails when running the tests.
```rust
let before = self.location();
self.update_location();
assert_eq!(before, self.location());
```

I also removed incorrect messages like "parent entity is not modified" and the unhelpful "Inserting a bundle in the children entities may change the parent entity's location if they were of the same archetype" which might lead people to think that's the *only* thing that can change the entity's location.

# Changelog
Added `EntityMut::world_scope`.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by nicopap** [comment]
In #5845, I had added a method to update the world safely: [`with_world_mut`](https://github.com/nicopap/bevy/blob/359ac8c9752c8f2e5bd78639ae63b7d78a57517f/crates/bevy_ecs/src/world/entity_ref.rs#L511-L523), but it never got merged. This allowed removing a lot of unsafe code.

Maybe you could add it? There were some discussion around it, you can find them there: https://github.com/bevyengine/bevy/pull/5845/files

**Review by nicopap — APPROVED** [review]
lgtm

File: crates/bevy_ecs/src/world/entity_ref.rs [hunk 1]
```diff
@@ -543,6 +543,12 @@ impl<'w> EntityMut<'w> {
         self.world
     }
 
+    /// Gives mutable access to this `EntityMut`'s [`World`] in a temporary scope.
```
Comment (nicopap) [inline crates/bevy_ecs/src/world/entity_ref.rs:546]: Might benefit from explaining why you'd want to use it.

Comment (devil-ira) [reply crates/bevy_ecs/src/world/entity_ref.rs:546]: It's the only way of safely getting a `&mut World` from an `EntityMut` without consuming `self`. Which sort of speaks for itself? 
I'm not sure what to add to the docs that could make that clearer (Aside from being very pedantic) :)

Comment (alice-i-cecile) [reply crates/bevy_ecs/src/world/entity_ref.rs:546]: You could discuss update_location here, but that's very gory-internals, and likely to go out of date in dangerous ways. If people care, they can read the code.

**Review by alice-i-cecile — APPROVED** [review]
bors r+

**Comment by alice-i-cecile** [comment]
Can you add a change log item for the new `world_scope` API? It's a nice little public usability improvement.

File: crates/bevy_ecs/src/world/entity_ref.rs (hunk 1 above)
Comment (nicopap) [reply crates/bevy_ecs/src/world/entity_ref.rs:546]: I was thinking: someone reading the doc doesn't necessarily see the source code. If they encounter this method, they may ask themselves why its here. But I agree with alice, it's really a minor thing and really depends on internals.

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**